test/elixir_bench_web/schema/schema_test.exs

```diff
@@ -79,30 +81,403 @@ defmodule ElixirBenchWeb.SchemaTest do
+      resp_body = context.conn
+        |> execute_query(query)
         |> Map.get(:resp_body)
-        |> Antidote.decode
+
+      Enum.each(required_fields, fn field ->
+        assert(
+          Regex.match?(~r/errors.*#{field}.*found null/, resp_body),
+          "Field '#{field}' not found in #{resp_body}"
+        )
+      end)
+    end
+
+    test "return slug error message when invalid slug is given", context do
+      query = """
+        repo (slug: "invalidSlug") {
+          name
+        }
+      """
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      error_message = Enum.at(resp["errors"], 0)
+
+      assert %{"message" => "invalid_slug"} = error_message
+    end
+  end
+
+  describe "benchmark query" do
+    test "fetch benchmark by name and repo slug", context do
+      benchmark = insert(:benchmark, %{name: "insert_mysql/insert_plain"})
+      repo = benchmark.repo
+
+      query = """
+        benchmark (
+          repo_slug: "#{repo.owner}/#{repo.name}",
+          name: "insert_mysql/insert_plain"
+        ) {
+          name
+        }
+      """
+
+      json_data =
+      %{
+        "data" => %{
+           "benchmark" => %{
+             "name" => "insert_mysql/insert_plain",
+           }
+         }
+       }
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      assert ^json_data = resp
+    end
+
+    test "return not found message when benchmark does not exist", context do
+      insert(:repo, %{name: "kernel", owner: "elixir-lang"})
+
+      query = """
+        benchmark (
+          repo_slug: "elixir-lang/kernel",
+          name: "Idont/exist"
+        ) {
+          name
+        }
+      """
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
 
       error_message = Enum.at(resp["errors"], 0)
 
-      refute resp["data"]["repo"]
       assert %{"message" => "not_found"} = error_message
     end
+
+    test "return not found when benchmark repo does not exist", context do
+      query = """
+        benchmark (
+          repo_slug: "Idont/exist",
+          name: "insert_mysql/insert_plain"
+        ) {
+          name
+        }
+      """
+
+      {:ok, resp} = context.conn
+        |> execute_query(query)
+        |> decode_response_body
+
+      error_message = Enum.at(resp["errors"], 0)
+
+      assert %{"message" => "not_found"} = error_message
+    end
+
+    test "return required fields message when missing fields", context do
+      required_fields = ["repoSlug", "name"]
+
+      query = """
+        benchmark {
+          name
+        }
+      """
+
+      resp_body = context.conn
+        |> execute_query(query)
+        |> Map.get(:resp_body)
+
+      Enum.each(required_fields, fn field ->
+        assert(
+          Regex.match?(~r/errors.*#{field}.*found null/, resp_body),
+          "Field '#{field}' not found in #{resp_body}"
```

Comment:
I know see where the more generic code further up comes from :) If we do it this way, I'd extract it into a helper method to use.

Also small thing - using `=~ instead of Regexp functions is better in my experience as you get better error messages

Reply:
I extracted it to a helper method and changed to use the ```=~``` match operator.